### PR TITLE
Replace the existing integration test for calculate your holiday entitlement flow

### PR DESCRIPTION
**Pull request by gclssvglx**
# What 

Replace the current MiniTest integration test with an RSpec version for the calculate your holiday entitlement flow.

# Why

We are in the process of replacing MiniTest tests with RSpec versions as this is the GDS preferred testing framework.

[Trello](https://trello.com/c/lForiKEa/658-add-rspec-test-for-calculate-your-holiday-entitlement)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    scenario "full-year" do
+      calculation_period calculation_periods[:full_year]
+      how_many_days_per_week 5
+      expect(page).to have_selector("p", text: statutory_entitlement(28))
+    end
+
+    scenario "starting" do
+      calculation_period calculation_periods[:starting]
+      what_is_your_starting_date(14, 3, Time.zone.now.year)
+      when_does_your_leave_year_start(2, 3, Time.zone.now.year)
```
Comment (theseanything) [inline spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb:173]: I think we can pass in a year directly here as "2020" (for all tests). I don't understand the benefit of freezing time and calling Time.now.

File: spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    "The statutory entitlement is #{hours} hours holiday."
+  end
+
+  def statutory_entitlement_compressed(hours, minutes)
+    "The statutory holiday entitlement is #{hours} hours and #{minutes} minutes holiday for the year. Rather than taking a day’s holiday it’s 8 hours and 0 minutes holiday for each day otherwise worked."
+  end
+
+  before do
+    # 2020 was a leap year
+    Timecop.freeze(Time.zone.local(2020))
```
Comment (theseanything) [inline spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb:150]: See comment below - I don't think we need to be freezing time anymore

**Review by theseanything — CHANGES_REQUESTED** [review]

File: spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb (hunk 1 above)
Comment (gclssvglx) [reply spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb]: Good point - have update to use `2020`.  Thanks

File: spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb (hunk 2 above)
Comment (gclssvglx) [reply spec/features/smart_answer/calculate_your_holiday_entitlement_flow_spec.rb]: Yes, absolutely right - excellent point.  Removed `Timecop`

**Review by theseanything — APPROVED** [review]

**Pull request merged**